docs/subnets/walkthrough-prompting.md

```diff
@@ -0,0 +1,87 @@
+
+- [Bittensor Building Blocks](../learn/bittensor-building-blocks).
+- [Anatomy of Incentive Mechanism](../learn/anatomy-of-incentive-mechanism).
+
+The below diagram shows a typical subnet with many subnet miners and subnet validators together executing the subnet incentive mechanism code. On the [Taostats explorer (select Subnet 01: Text Prompting from the top menu)](https://taostats.io/) the **Metagraph** view for this Subnet 1: Prompting shows the performance details for each subnet miner and validator. 
+
+For easier understanding, in this document we will focus on how a **single** subnet validator interacts with **multiple** subnet miners in this subnet. 
+
+<center id="bittensor-img">
+<ThemedImage
+alt="1-Prompting Walkthrough"
+sources={{
+    light: useBaseUrl('/img/docs/1-prompting-subnet-walkthrough.svg'),
+    dark: useBaseUrl('/img/docs/1-prompting-subnet-walkthrough.svg'),
+  }}
+style={{width: 600}}
+/>
+</center>
+
+## Subnet 1 summary
+
+See the below diagram showing a high-level view of how this Prompting subnet works. 
+
+<center>
+<ThemedImage
+alt="Prompting Subnet 1 Big Picture"
+sources={{
+    light: useBaseUrl('/img/docs/2-prompting-subnet-high-level.svg'),
+    dark: useBaseUrl('/img/docs/2-prompting-subnet-high-level.svg'),
+  }}
+style={{width: 750}}
+/>
+</center>
+
+The numbered items in the below description correspond to the numbered sections in the above diagram:
+
+1. The subnet validator sends a **challenge** simultaneously to multiple subnet miners. This step constitutes the **query** from the subnet validator to subnet miners. A challenge is a prompt sent to subnet miners in such a way that:
+   - This prompt is in a style and tone that is similar to humans.
+   - This prompt drives the conversation between the subnet validator and the subnet miners in order to reach a pre-defined goal. 
+
+    See below section for how a challenge is generated. 
+2. The subnet miners respond to the subnet validator after performing the challenge **task**. 
+3. The subnet validator then **scores** each subnet miner by comparing the subnet miner's response with a locally generated **reference** answer. The subnet validator uses this reference as the ground truth for this step. 
+4. Finally, the subnet validator **sets the weights** for the subnet miners by sending the weights to the blockchain. In the Bittensor blockchain the Yuma Consensus allocates the rewards to the participating subnet miners and subnet validators. 
+
+:::tip Use of large language models 
+In this subnet both the subnet validator and the subnet miners use large language models (LLMs) to create the challenges (subnet validator) and respond to the prompts (subnet miners).
+:::
+
+## Challenge generation
+
+The subnet validator composes a challenge based on whether the task is answering questions, summarizing a given text, debugging code, solve mathematics problems. The challenge-creation works like this:
```

Comment:
The motivation behind using multiple tasks is severalfold:
1. It constantly benchmarks the capabilities of the miners across a broad range of challenging tasks that are common use-cases. This can be used to route queries to specialized miners, thereby providing an effective mixture of experts system.
2. It is also a precursor to an inter-subnet bridging mechanism that will enable SN1 to 'reach out' into other subnets are access the useful work that can be provided by them. Miners must become adept at using tools and APIs in order to fulfill validation tasks. We are building an API layer for inter-subnet communication, which is a natural extension of 'agentic' models.